src/inference/enumerate.ad.js

```diff
@@ -67,21 +93,33 @@ module.exports = function(env) {
     // Find support of this distribution:
     if (dist.isContinuous || !dist.support) {
       console.error(dist);
-      throw new Error('Enumerate can only be used with distributions that have finite support.');
+      return 'Enumerate can only be used with distributions that have finite support.';
     }
     var supp = dist.support();
 
     // Check that support is non-empty
     if (supp.length === 0) {
       console.error(dist);
-      throw new Error('Enumerate encountered a distribution with empty support!');
+      return 'Enumerate encountered a distribution with empty support.';
     }
-
     return supp;
   };
 
   Enumerate.prototype.sample = function(store, k, a, dist) {
     var support = getSupport(dist);
+    if (_.isString(support)) {
```

Comment:
I tried to do so at first. But it seems the error function called in `getSupport()` like `return env.coroutine.error('...');` doesn't interrupt the execution of enumeration? The enumeration keeps running rather than return error msg to header.wppl. I'm a little confused.